### ISSUE TITLE
bootstrapping container with lxc.present

### ISSUE THREAD

**Issue by thepauleh** [saltstack/salt#25236]
Hi,

I've been trying to get containers working using lxc.present and also salt cloud. My issue doing this with lxc.present is that theres no minion configured at the end of the process.

```
{{ ROLE }}-lxc:
  lxc.present:
        - name: {{ ROLE }}-lxc
        - running: True
        - template: centos
```

Using this technique, the container can be started - but has no minion.

If i try to use the salt cloud approach, (cloud.profile) then the container fails to start due to bad gateway config (i'm trying to go for a dhcp config inside of vagrant vm) - I have only defined the master ip & port, and I was hoping the rest would be treated the same as the lxc.present for network config & 'just work':

```
my-lxc:
  # the provider to use
  provider: my-lxc
  lxc_profile:
      template: centos
  minion:
     master: 10.0.2.15
     master_port: 4506
```

Specifically:
```
lxc-start: start.c: lxc_spawn: 817 failed to find gateway addresses
lxc-start: start.c: __lxc_start: 1080 failed to spawn 'pauls1-lxc'
lxc-start: lxc_start.c: main: 342 The container failed to start.
lxc-start: lxc_start.c: main: 346 Additional information can be obtained by setting the --logfile and --logpriority options.
```
Has anyone been here before?

Looks like this issue is kinda the same for the 'cheat/lxc.present' approach:
https://github.com/saltstack/salt/issues/24752

**Comment by basepi** [comment]
This does look very similar to #24752. However, since this one has more information, I think I'm going to actually close that one in favor of this, and mark is as a duplicate. Thanks!

**Comment by basepi** [comment]
On second thought, this one uses salt-cloud and that one uses the state module, so I'm going to keep both open.

**Comment by EvaSDK** [comment]
Previous versions of salt tried to set gateway to auto for your container even if using dhcp (https://github.com/saltstack/salt/blob/2014.7/salt/modules/lxc.py#L201). LXC does not like that gateway is defined but not the rest of network information so it fails to start. It should be solved with 2015.5.3 as I have managed to bootstrap some LXC slaves with it.

**Comment by thepauleh** [comment]
Ok, great news regarding salt-cloud for this, it seems the way i was trying to set the network profile wasn't great, and having a VirtualBox host i need virbr0 as my link (which was being ignored, probs due to default interfaces file as mentioned in the docs).

Overrode it with: network_profile: {'eth0': {'link': 'virbr0'}}

However, now at this monster:
The minion function caused an exception: Traceback (most recent call last): File "/usr/lib/python2.6/site-packages/salt/minion.py", line 1161, in _thread_return return_data = func(*args, **kwargs) File "/usr/lib/python2.6/site-packages/salt/modules/lxc.py", line 1566, in init bootstrap_args=bootstrap_args) File "/usr/lib/python2.6/site-packages/salt/modules/lxc.py", line 3198, in bootstrap bootstrap=bootstrap_url) File "/usr/lib/python2.6/site-packages/salt/modules/config.py", line 401, in gather_bootstrap_script ret = salt.utils.cloud.update_bootstrap(__opts__, url=bootstrap) File "/usr/lib/python2.6/site-packages/salt/utils/cloud.py", line 2325, in update_bootstrap req = requests.get(url) File "/usr/lib/python2.6/site-packages/requests/api.py", line 55, in get return request('get', url, **kwargs) File "/usr/lib/python2.6/site-packages/requests/api.py", line 44, in request return session.request(method=method, url=url, **kwargs) File "/usr/lib/python2.6/site-packages/requests/sessions.py", line 279, in request resp = self.send(prep, stream=stream, timeout=timeout, verify=verify, cert=cert, proxies=proxies) File "/usr/lib/python2.6/site-packages/requests/sessions.py", line 374, in send r = adapter.send(request, **kwargs) File "/usr/lib/python2.6/site-packages/requests/adapters.py", line 213, in send raise SSLError(e) SSLError: hostname 'bootstrap.saltstack.com' doesn't match either of '*.wpengine.com', 'wpengine.com'

Closing this as this error above is covered by: https://github.com/saltstack/salt/issues/25236
